funcx_endpoint/funcx_endpoint/cli.py

```diff
@@ -0,0 +1,226 @@
+        return click.get_current_context().ensure_object(CommandState)
+
+
+def get_cli_endpoint() -> Endpoint:
+    # this getter creates an Endpoint object from the CommandState
+    # it takes its various configurable values from the current CommandState
+    # as a result, any number of CLI options may be used to tweak the CommandState
+    # via callbacks, and the Endpoint will only be constructed within commands which
+    # access the Endpoint via this getter
+    state = CommandState.ensure()
+
+    endpoint = Endpoint(funcx_dir=state.endpoint_config_dir, debug=state.debug)
+
+    # ensure that configs exist
+    if not os.path.exists(endpoint.funcx_config_file):
+        log.info(
+            "No existing configuration found at %s. Initializing...",
+            endpoint.funcx_config_file,
+        )
+        endpoint.init_endpoint()
+
+    return endpoint
+
+
+def common_options(f):
+    def debug_callback(ctx, param, value):
+        if not value or ctx.resilient_parsing:
+            return
+
+        state = CommandState.ensure()
+        state.debug = True
+
+    f = click.option(
+        "--debug",
+        is_flag=True,
+        hidden=True,
+        expose_value=False,
+        callback=debug_callback,
+        is_eager=True,
+    )(f)
+    f = click.help_option("-h", "--help")(f)
+    return f
+
+
+def name_option(f):
+    return click.option(
+        "-n", "--name", help="endpoint name", show_default=True, default="default"
+    )(f)
+
+
+def config_dir_callback(ctx, param, value):
+    if value is None or ctx.resilient_parsing:
+        return
+    state = CommandState.ensure()
+    state.endpoint_config_dir = value
+
+
+@click.group("funcx-endpoint")
+@common_options
+@click.option(
+    "-c",
+    "--config-dir",
+    default=None,
+    help="override default config dir",
+    callback=config_dir_callback,
+    expose_value=False,
+)
+def app():
+    # the main command group body runs on every command, so the block below will always
+    # execute
+    state = CommandState.ensure()
+    setup_logging(debug=state.debug)
+
+
+@app.command("version")
+@common_options
+def version_command():
+    """Show the version of funcx-endpoint"""
+    import funcx_endpoint
+
+    click.echo(f"FuncX endpoint version: {funcx_endpoint.__version__}")
+
+
+@app.command(name="configure", help="Configure an endpoint")
+@click.option("--endpoint-config", default=None, help="a template config to use")
+@name_option
+@common_options
+def configure_endpoint(*, name: str, endpoint_config: str | None):
+    """Configure an endpoint
+
+    Drops a config.py template into the funcx configs directory.
+    The template usually goes to ~/.funcx/<ENDPOINT_NAME>/config.py
+    """
+    endpoint = get_cli_endpoint()
+    endpoint.configure_endpoint(name, endpoint_config)
+
+
+@app.command(name="start", help="Start an endpoint by name")
+@name_option
+@click.option(
+    "--endpoint-uuid", default=None, help="The UUID for the endpoint to register with"
```

Comment:
I believe this kwarg option doesn't need to be specified right now and is taken as a positional arg. Adding this would need a minor fix to the invocation used in kube-endpoint's `boot.sh`

Reply:
Much more strongly than thinking that `name` should be an option vs positional arg, this should _definitely_ be an option.
I see that this was recently changed, as it's `--endpoint-id` in `main`: https://github.com/funcx-faas/funcX/blob/9fd357d96d0d7a68fe774b3567498a1dab324355/helm/boot.sh#L11

Simple then: that change should be reverted and I'll add that to this PR.

Reply:
I've fixed the boot.sh script back to what it was in `main` (and, I'll note, what it was when this PR was opened) in a rebased and updated version.
If you're satisfied with that, let's mark this resolved.

Reply:
Nice!